### PR TITLE
Validation pipeline plot isoscape

### PR DESCRIPTION
**Pull request by erickzul**

### CODE REVIEW

File: validation_pipeline.ipynb [hunk 1]
```diff
@@ -7,7 +7,7 @@
         "colab_type": "text"
```
Comment (rothn) [inline validation_pipeline.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #10.    TEST_SET_FILENAME = 'canonical/uc_davis_no_partition_test_random_grouped.csv' #@param
</pre><p>I saw this in Greta's notebook too. What does "no_partition" mean?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by erickzul** [comment]
Thank you! This PR should be listed as "Draft", which means it is not ready for review. This is a great start, though!

File: validation_pipeline.ipynb (hunk 1 above)
Comment (benwulfe) [inline validation_pipeline.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #13.      predictions.dropna(subset=['fraud', 'fraud_p_value'], inplace=True)
</pre><p>why do we drop these?  if we get a nan p-value, is something is wrong with the input set?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (erickzul) [reply validation_pipeline.ipynb:7]: No, it just means that we don't have enough samples for that grouping (i.e. just one measurement for the element). When it's a combined p value (Carbon, Oxygen and Nitrogen). It means that no element had enough measurements (>= 1)

**Review by benwulfe — APPROVED** [review]
looks good.  in a future PR, consider merging some of the code blocks to ease execution

File: validation_pipeline.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rothn) [reply validation_pipeline.ipynb:7]: <p>Resolved-- no longer an issue</p>

Comment (rothn) [reply validation_pipeline.ipynb:7]: <p>Ok, now I'm confused. Can we define "grouping" somewhere? Does this refer to measurements from an individual sample?</p>

**Review by rothn — CHANGES_REQUESTED** [review]
I resolved one comment that is no longer relevant. Please let me know regarding the other comments I've left on this PR.

File: validation_pipeline.ipynb (hunk 1 above)
Comment (erickzul) [reply validation_pipeline.ipynb:7]: It means that this is the entire dataset being used for evaluation.

[the remark this answers is not in the document]
Comment (erickzul) [reply validation_pipeline.ipynb:7]: This has since last review request been removed.

[the remark this answers is not in the document]
Comment (erickzul) [reply validation_pipeline.ipynb:7]: We do joins because our current implementation only calculates metrics with ungrouped data. We're changing this here: https://github.com/tnc-br/ddf_common/pull/30

The input test data is grouped, unfortunately, so we "reverse" that. I will change this to the grouped version once that is checked in.

Comment (rothn) [reply validation_pipeline.ipynb:7]: But then it also says "test". So is this the entire unpartitioned dataset or the test set?

[the remark this answers is not in the document]
Comment (rothn) [reply validation_pipeline.ipynb:7]: <p>I still don't think I understand what you mean here. How is it possible to reverse aggregation?</p><p> </p><p>Also, it seems like this first line is a no-op. Is that intentional?</p><p><code></code><code></p><p>&nbsp;test_dataset = pd.concat([real, fake_sample], axis=1, join='outer')</p><p>&nbsp;test_dataset = real.append(fake_sample, ignore_index=True)</p><p></code><code></code></p><p> </p>

[the remark this answers is not in the document]
Comment (erickzul) [reply validation_pipeline.ipynb:7]: Updated to set random seed.

[the remark this answers is not in the document]
Comment (erickzul) [reply validation_pipeline.ipynb:7]: Great idea! Plotted.

Comment (erickzul) [reply validation_pipeline.ipynb:7]: Yes, the grouping here is [code, long, lat] which is what we define as a group of samples.

Comment (erickzul) [reply validation_pipeline.ipynb:7]: Entire unpartitioned dataset. If we really need to, I can change the name. But it may be a small detail if we're not planning to continue evaluation on this set altogether.

[the remark this answers is not in the document]
Comment (erickzul) [reply validation_pipeline.ipynb:7]: You take the unique code, long, lat codings and join them with the ungrouped dataset. This will result in a dataset that has the original 5 samples but now they have the aggregate mean and variance columns.

Comment (rothn) [reply validation_pipeline.ipynb:7]: Discussed online: This came from the request of evaluating on the training set. Now, the data ingestion script does preprocessing and grouping but no partitioning. This is a placeholder now and will be changed to the holdout set moving forward, which we have not defined yet.

Mitigation strategy: Add a TODO before submitting.

[the remark this answers is not in the document]
Comment (rothn) [reply validation_pipeline.ipynb:7]: AI: Delete the first line

[the remark this answers is not in the document]
Comment (rothn) [reply validation_pipeline.ipynb:7]: Additional context: Differences were due to using different eval datasets. We have checked results on both now and confirmed on our own and with Martinelli's analysis that present direction (VI model, no boosting, all the inputs) makes sense.

**Review by rothn — APPROVED** [review]
Approved, provisionally given a TODO is added not to test on the training set.
